tests/rptest/tests/custom_topic_assignment_test.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2020 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import random
+
+from ducktape.mark.resource import cluster
+from ducktape.utils.util import wait_until
+from rptest.clients.python_librdkafka import PythonLibrdkafka
+from rptest.clients.rpk import RpkTool
+from rptest.tests.redpanda_test import RedpandaTest
+from confluent_kafka.admin import NewTopic
+from confluent_kafka.error import KafkaException, KafkaError
+
+from rptest.clients.types import TopicSpec
+from rptest.clients.kafka_cli_tools import KafkaCliTools
+
+
+class CustomTopicAssignmentTest(RedpandaTest):
+    def __init__(self, test_context):
+        super(CustomTopicAssignmentTest,
+              self).__init__(test_context=test_context, num_brokers=5)
+
+    def create_and_validate(self, name, custom_assignment):
+        self.redpanda.logger.info(
+            f"creating topic {name} with {custom_assignment}")
+        cli = KafkaCliTools(self.redpanda)
+        rpk = RpkTool(self.redpanda)
+
+        cli.create_topic_with_assignment(name, custom_assignment)
+
+        def replica_matches():
+            replicas_per_partition = {}
+            for p in list(rpk.describe_topic(name)):
```

Comment:
if `describe_topic` returns a `map` object the iteration here should unpack that without needing to materialize the whole list with `list()`